pkg/controller/assign/assign_controller.go

```diff
@@ -0,0 +1,141 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assignment
```

Comment:
nit: `package assign` to match the directory name?

Reply:
This is to fix the lint error:
```
type name will be used as assign.AssignReconciler by other packages, and that stutters; consider calling this Reconciler
```

Reply:
Could we call it `Reconciler` instead of renaming the package?

Otherwise we'll need to do something like this:

`assign "github.com/open-policy-agent/gatekeeper/pkg/controller/assign"`

every time we import the controller.

Reply:
We could. By the way I think the same lint issue is why we also have `package assignmentmetadata`

Reply:
ah, that one slipped by me. We should probably change that as well.

@mmirecki @fedepaol any objections to changing the name of the "assignmentmetadata" package to match the directory name?

Reply:
Not at all. Sounds legit to me